astropy/modeling/tabular.py

```diff
@@ -0,0 +1,175 @@
+        "nearest", and "splinef2d". "splinef2d" is only supported for
+        2-dimensional data. Default is "linear".
+    bounds_error : bool, optional
+        If True, when interpolated values are requested outside of the
+        domain of the input data, a ValueError is raised.
+        If False, then `fill_value` is used.
+    fill_value : float, optional
+        If provided, the value to use for points outside of the
+        interpolation domain. If None, values outside
+        the domain are extrapolated.  Extrapolation is not supported by method
+        "splinef2d".
+
+    Returns
+    -------
+    value : ndarray
+        Interpolated values at input coordinates.
+
+    Raises
+    ------
+    ImportError
+        Scipy is not installed.
+
+    Notes
+    -----
+    Uses ``scipy.interpolate.interpn``
+
+    """
+
+    linear = False
+    fittable = False
+
+    standard_broadcasting = False
+    outputs = ('y',)
+
+    lookup_table = abc.abstractproperty()
+
+    _is_dynamic = True
+
+    _id = 0
+
+    def __init__(self, points=None, lookup_table=None, method='linear',
+                 bounds_error=True, fill_value=np.nan, **kwargs):
+
+        super(Tabular, self).__init__(**kwargs)
+        if lookup_table is not None:
+            lookup_table = np.asarray(lookup_table)
+            if self.lookup_table.ndim != lookup_table.ndim:
+                raise ValueError("lookup_table should be an array with "
+                                 "{0} dimensions".format(self.lookup_table.ndim))
+            self.lookup_table = lookup_table
+        if points is None:
+            self._points = tuple(np.arange(x, dtype=np.float)
+                                 for x in self.lookup_table.shape)
+        else:
+            if self.lookup_table.ndim == 1:
+                self._points = (points,)
+            else:
+                self._points = points
+            if len(self._points) != self.lookup_table.ndim:
+                raise ValueError("Expected grid points in "
+                                 "{0} directions, got {1}".format(self.lookup_table.ndim,
+                                                                  len(self._points)))
+
+        self.bounds_error = bounds_error
+        self.method = method
+        self.fill_value = fill_value
+
+    @property
+    def points(self):
+        return self._points
+
+    def evaluate(self, *inputs):
+        """
+        Return the interpolated values at the input coordinates.
+
+        Parameters
+        ----------
+        inputs : list of scalars or ndarrays
+            Input coordinates. The number of inputs must be equal
+            to the dimensions of the lookup table.
+        """
+        inputs = np.array(inputs[: self.n_inputs]).T
+        if not has_scipy:
+            raise ImportError("This model requires scipy >= v0.14")
+        return interpn(self.points, self.lookup_table, inputs,
```

Comment:
When I try to use this with `points` in descending order (which PySynphot and `interp1d` both support), I get:

```python
ValueError: The points in dimension 0 must be strictly ascending
```

Is there a way to get around this without having to re-order the data myself? I could overwrite my subclass' `evaluate()` using `interp1d` but that kind of defeats the purpose of subclassing.